src/Filesystem/Flag/Manager.php

```diff
@@ -37,6 +37,11 @@ class Manager
      */
     public const FLAG_IGNORE_SPLIT_DB = 'ignore_split_db';
 
+    /**
+     * Used to mark that env.php file does not exist at the beginning of deployment
+     */
+    public const FLAG_ENV_FILE_ABSENCE = 'env_file_absence';
```

Comment:
My understanding was that the flags were useful to communicate between build & deploy stages, or between multiple deployments. Why are we using them here were everything is happening within a single deployment?

Reply:
good question. Actually, we can understand whether env.php was removed or not only at the beginning of deploy phase, because already on the second step `cache` https://github.com/magento/ece-tools/blob/develop/scenario/deploy.xml#L8 we write data to env.php file and if it is absent we create this file with some part of data. But steps which should react on file absence go much later and at that moment file always exist